ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AvailableStorageTable.test.tsx

```diff
@@ -273,6 +273,45 @@ describe("AvailableStorageTable", () => {
     expect(wrapper.find("AddLogicalVolume").exists()).toBe(true);
   });
 
+  it("can open the edit logical volume form", () => {
+    const volumeGroup = diskFactory({ type: DiskTypes.VOLUME_GROUP });
+    const logicalVolume = diskFactory({
+      available_size: MIN_PARTITION_SIZE + 1,
+      parent: { id: volumeGroup.id, type: volumeGroup.type, uuid: "vg0" },
+      type: DiskTypes.VIRTUAL,
+    });
+    const state = rootStateFactory({
+      machine: machineStateFactory({
+        items: [
+          machineDetailsFactory({
+            disks: [volumeGroup, logicalVolume],
+            system_id: "abc123",
+          }),
+        ],
+        statuses: machineStatusesFactory({
+          abc123: machineStatusFactory(),
+        }),
+      }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <AvailableStorageTable canEditStorage systemId="abc123" />
+      </Provider>
+    );
+
+    wrapper.find("TableMenu button").at(1).simulate("click");
+    wrapper
+      .findWhere(
+        (button) =>
+          button.name() === "button" &&
+          button.text() === "Edit logical volume..."
```

Comment:
This is coupling the functionality tests with appearance. Can you target a class so if we can the text in the button the tests don't fail?